app/components/Messages.tsx

```
@@ -231,21 +232,39 @@ export const Messages = ({
+  const toolGroupMountStateRef = useRef<{
+    chatId: string;
+    hasCommittedTimeline: boolean;
+    seenToolGroupIds: ReadonlySet<string>;
+  }>({
+    chatId,
+    hasCommittedTimeline: false,
+    seenToolGroupIds: new Set(),
+  });
+  const rawTimelineRows = useMemo(() => {
+    const toolGroupMountState =
+      toolGroupMountStateRef.current.chatId === chatId
```

ESLint

[error] Error: Cannot access refs during render

React refs are values that are not needed for rendering. Refs should only be accessed outside of render, such as in event handlers or effects. Accessing a ref value (the `current` property) during render can cause your component not to update as expected (https://react.dev/reference/react/useRef).

/home/jailuser/git/app/components/Messages.tsx:246:7
  244 |   const rawTimelineRows = useMemo(() => {
  245 |     const toolGroupMountState =
> 246 |       toolGroupMountStateRef.current.chatId === chatId
      |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Cannot access ref value during render
  247 |         ? toolGroupMountStateRef.current
  248 |         : {
  249 |             hasCommittedTimeline: false,

(react-hooks/refs)